test/e2e/far_e2e_test.go

```diff
@@ -0,0 +1,171 @@
+			})
+			It("should reboot node", func() {
+				By("checking node's boot time")
+				Eventually(func() (time.Time, error) {
+					bootTime, err := farUtils.GetBootTime(clientSet, validNodeName, testNamespace, log)
+					if bootTime != nil && err == nil {
+						log.Info("got boot time", "time", *bootTime)
+						return *bootTime, nil
+					}
+					log.Error(err, "failed to get boot time")
+					return time.Time{}, err
+				}, timeout, pollInterval).Should(
+					BeTemporally(">", testStart),
+				)
+			})
+		})
+	})
+})
+
+// createFAR assign the input to FenceAgentsRemediation object, create CR it with offset, and return the CR object
+func createFAR(nodeName string, agent string, sharedparameters map[v1alpha1.ParameterName]string, nodeparameters map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string) *v1alpha1.FenceAgentsRemediation {
+	far := &v1alpha1.FenceAgentsRemediation{
+		ObjectMeta: metav1.ObjectMeta{Name: nodeName, Namespace: testNamespace},
+		Spec: v1alpha1.FenceAgentsRemediationSpec{
+			Agent:            agent,
+			SharedParameters: sharedparameters,
+			NodeParameters:   nodeparameters,
+		},
+	}
+	ExpectWithOffset(offsetExpect, k8sClient.Create(context.Background(), far)).ToNot(HaveOccurred())
+	return far
+}
+
+// deleteFAR delete the CR with offset
+func deleteFAR(far *v1alpha1.FenceAgentsRemediation) {
+	EventuallyWithOffset(offsetExpect, func() error {
+		err := k8sClient.Delete(context.Background(), far)
+		if apiErrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}, timeout, pollInterval).ShouldNot(HaveOccurred(), "failed to delete far")
+}
+
+// checkFarLogs get the FAR pod and check whether its logs has logString
+func checkFarLogs(node *corev1.Node, logString string) {
+	By("checking logs")
+	pod, err := getFenceAgentsPod(testNamespace)
+	if err != nil {
+		fmt.Print(err)
+	}
+	ExpectWithOffset(offsetExpect, pod).ToNot(BeNil())
+
+	EventuallyWithOffset(offsetExpect, func() string {
+		var err error
+		logs, err := farUtils.GetLogs(clientSet, pod, "manager")
+		if err != nil {
+			log.Error(err, "failed to get logs. Might try again")
+			return ""
+		}
+		return logs
+	}, timeout/2, pollInterval).Should(ContainSubstring(logString))
+}
+
+// getFenceAgentsPod fetches the FAR pod based on FAR's label and namespace
+func getFenceAgentsPod(namespace string) (*corev1.Pod, error) {
+	pods := new(corev1.PodList)
+	podLabelsSelector, _ := metav1.LabelSelectorAsSelector(
+		&metav1.LabelSelector{MatchLabels: farController.FaPodLabels})
+	options := client.ListOptions{
+		LabelSelector: podLabelsSelector,
+		Namespace:     namespace,
+	}
+	if err := k8sClient.List(context.Background(), pods, &options); err != nil {
+		return nil, err
+	}
+	if len(pods.Items) == 0 {
+		podNotFoundErr := &apiErrors.StatusError{ErrStatus: metav1.Status{
+			Status: metav1.StatusFailure,
+			Code:   http.StatusNotFound,
+			Reason: metav1.StatusReasonNotFound,
+		}}
+		return nil, podNotFoundErr
+	}
+	return &pods.Items[0], nil
+}
```

Comment:
I have copy paste the function from the [controller](https://github.com/medik8s/fence-agents-remediation/blob/main/controllers/fenceagentsremediation_controller.go#L112-L137), since I don't have the reconciler object in far_e2e_test.go.
I can call this function from  far_e2e_test.go but it would require removing the reconciler which is needed for listing the objects in the controller

Reply:
No longer true after #42